tagging/tag_service.py

```diff
@@ -134,32 +134,83 @@ def add_tags(self, tags: List[Tag], asset_uid: str) -> Dataset:
         asset_tags_uid : str
             ID of the existing tag set to update
 
-
         Returns
         ----------
-        Dataset
-            The updated tag set (full asset tag object)
+        added_tags_uid : List[str]
+            List of the UIDs of the added tags
         """
+        added_tags_uid = []
+        # If no tags were sent, returns empty list
+        if tags is None:
+            return added_tags_uid
 
-        doc_tags = self._collection_dataset.find_one({'uid': asset_uid})
-        if not doc_tags:
+        asset = self._collection_dataset.find_one({'uid': asset_uid})
+        if not asset:
             raise DatasetNotFound(f"no asset with id: {asset_uid}")
 
-        if not doc_tags.get('tags'):
-            doc_tags['tags'] = []
+        if asset['tags'] is None:
+            self._collection_dataset.update_one(
+                {'uid': asset_uid},
+                {'$set': {'tags': []}})
 
-        for tag in tags:
-            doc_tags['tags'].append(tag.dict())
+        tags_dict = []
+        for user_tag in tags:
+            tag = Tag(**user_tag.dict())
+            tags_dict.append(tag.dict())
+            added_tags_uid.append(tag.uid)
 
-        # Takes a asset tag uid key to find one to change and then passes in
-        # new tag array
+        print(tags_dict)
```

Comment:
Let's remove print statements. This code should make better use of logger, for another PR.